library/ssl_tls.c

```diff
@@ -5199,6 +5203,23 @@ int mbedtls_ssl_handshake_step( mbedtls_ssl_context *ssl )
     }
 #endif
 
+    if( ret != 0 )
+    {
+        /* handshake_step return error. And it is same
+         * with alert_reason.
+         */
+        int alert_ret;
+        alert_ret = mbedtls_ssl_handle_pending_alert( ssl );
+        if( alert_ret != 0 )
```

Comment:
```suggestion
        if( ssl->send_alert )
        {
            ret = mbedtls_ssl_handle_pending_alert( ssl );
            goto cleanup;
        }
```
That looks clearer and simpler to understand to me like this.